### PR TITLE
Avoid collisions between essential packages

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
Fix remaining collision between essential packages.
See issue #55886.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/os-specific/linux/procps-ng/default.nix [hunk 1]
```diff
@@ -46,7 +46,7 @@ stdenv.mkDerivation rec {
   meta = {
     homepage = https://gitlab.com/procps-ng/procps;
     description = "Utilities that give information about processes using the /proc filesystem";
-    priority = 10; # less than coreutils, which also provides "kill" and "uptime"
+    priority = 11; # less than coreutils, which also provides "kill" and "uptime"
```
Comment (JohnAZoidberg) [inline pkgs/os-specific/linux/procps-ng/default.nix]: Coreutils has 10. Which wasn't less before but certainly isn't now.
Does that cause conflicts?

**Comment by JohnAZoidberg** [comment]
The priority thing is really not optimal... We should have relative priorities between packages.

**Comment by grahamc** [comment]
On 19.03, `nix-build -E 'with (import <nixpkgs> {}); buildEnv { name = "test"; paths = [coreutils procps]; }'` creates a environment where `kill` and `uptime` come from `coreutils`.  This feels backwards to me, since procps seems to be the more specific package, but :man_shrugging: @rnhmjoj I'm assuming this PR retains this behavior?

**Comment by rnhmjoj** [comment]
> where kill and uptime come from coreutils

Note that on master kill it's actually provided by util-linux.

>  I'm assuming this PR retains this behavior?

Yes, it does. I opened this PR just to fix those annoying warnings, nothing more.
Anyway, I agree that it would make more sense for procps to override and coreutils.

**Pull request merged**